packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+	 * or focuses something not part of the editing experience of that node
+	 */
+	function onBlur(event: React.FocusEvent<HTMLDivElement>): void {
+		/**
+		 * This allows the parent element to share focus with children
+		 * so the blur event only triggers if focus is outside of div and children
+		 */
+		if (!event.currentTarget.contains(event.relatedTarget)) {
+			setEditProp(undefined);
+		}
+	}
+
+	/**
+	 * Updates {@link editType} and {@link editingComponent} based on the option selected from the dropdown
+	 * If the option selected is null or undefined it will automatically send the message to change its value
+	 * and will only display the dropdown with either "null" or "undefined"
+	 * @param data - The option selected from the dropdown
+	 */
+	const onOptionSelect: DropdownProps["onOptionSelect"] = (event, data) => {
+		if (data.optionText === undefined) {
+			throw new Error("Invalid option text from dropdown");
+		}
+
+		if (data.optionText === "undefined") {
+			submitChange(undefined);
+		}
+
+		if (data.optionText === "null") {
+			// We need to support users waiting to use "null" as a value
+			// eslint-disable-next-line unicorn/no-null
+			submitChange(null);
+		}
+
+		let newValue: EditData;
+		if (editProp !== undefined) {
+			newValue = data.optionText === editProp.type ? editProp.value : "";
+		} else {
+			newValue = data.optionText === typeof node.value ? node.value : "";
+		}
+		setEditProp({
+			type: data.optionText,
+			value: newValue,
+		});
+	};
+
+	const options = node.editProps?.editTypes === undefined ? allEdits : node.editProps?.editTypes;
+
+	function getEditType(): string {
+		if (editProp === undefined) {
+			if (node.value === null) {
+				return "null";
+			}
+			return typeof node.value;
+		} else {
+			if (editProp.value === null) {
+				return "null";
+			}
+			return editProp.type;
+		}
+	}
+	let innerView: React.ReactElement;
+	switch (getEditType()) {
+		case "string":
+			innerView = (
+				<EditableInputComponent
+					node={node}
+					editProp={editProp}
+					setEditProp={setEditProp}
+					submitChange={submitChange}
+					inputType={"string"}
+				/>
+			);
+			break;
+		case "number":
+			innerView = (
+				<EditableInputComponent
+					node={node}
+					editProp={editProp}
+					setEditProp={setEditProp}
+					submitChange={submitChange}
+					inputType={"number"}
+				/>
+			);
+			break;
+		case "boolean":
+			innerView = (
+				<EditableBooleanComponent
+					node={node}
+					setEditProp={setEditProp}
+					editProp={editProp}
+					submitChange={submitChange}
+				/>
+			);
+			break;
+		case "undefined":
+			innerView = <i>: undefined</i>;
+			break;
+
+		case "null":
+			innerView = <i>: null</i>;
+			break;
+
+		default:
+			throw new Error("Unrecognized edit type.");
+	}
+
+	const styles = useStyles();
+
+	return (
+		<div
+			className={editProp !== undefined ? styles.isEditingStyle : styles.isNotEditingStyle}
```

Comment:
Nit: I usually recommend using the positive equality check when you're doing an if/else. I.e.
```suggestion
			className={editProp === undefined ? styles.isNotEditingStyle : styles.isEditingStyle }
```